src/CFamily.UnitTests/CMake/CompilationDatabaseLocatorTests.cs

```diff
@@ -212,24 +237,72 @@ public void Locate_HasCMakeSettingsFile_ConfiguredPathHasParameters_ParametersRe
             buildConfigProvider ??= Mock.Of<IBuildConfigProvider>();
             logger ??= Mock.Of<ILogger>();
             fileSystem ??= new FileSystem();
+            macroEvaluationService ??= Mock.Of<IMacroEvaluationService>();
 
-            return new CompilationDatabaseLocator(folderWorkspaceService.Object, buildConfigProvider, cMakeSettingsProvider, fileSystem, logger);
+            return new CompilationDatabaseLocator(folderWorkspaceService.Object, buildConfigProvider, cMakeSettingsProvider, macroEvaluationService, fileSystem, logger);
         }
 
-        private IBuildConfigProvider CreateConfigProvider(string activeConfiguration)
+        private static IBuildConfigProvider CreateConfigProvider(string activeConfiguration)
         {
             var provider = new Mock<IBuildConfigProvider>();
             provider.Setup(x => x.GetActiveConfig(It.IsAny<string>())).Returns(activeConfiguration);
             
             return provider.Object;
         }
 
-        private static Mock<ICMakeSettingsProvider> CreateCmakeSettingsProvider(CMakeSettingsSearchResult result)
+        private static Mock<ICMakeSettingsProvider> CreateCMakeSettingsProvider(string rootDirectory, CMakeSettingsSearchResult result)
         {
             var cmakeSettingsProvider = new Mock<ICMakeSettingsProvider>();
-            cmakeSettingsProvider.Setup(x => x.Find(RootDirectory)).Returns(result);
+            cmakeSettingsProvider.Setup(x => x.Find(rootDirectory)).Returns(result);
 
             return cmakeSettingsProvider;
         }
+
+        private static IMacroEvaluationService CreatePassthroughMacroEvaluator()
+        {
+            Func<string, string, string, string> passthrough = (input, _, _) => input;
+
+            var macroEvaluator = new Mock<IMacroEvaluationService>();
+            macroEvaluator.Setup(x => x.Evaluate(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>()))
+                .Returns(passthrough);
+
+            return macroEvaluator.Object;
+        }
+
+        /// <summary>
```

Comment:
There is still a lot of boilerplate setup required to put the locator in a state where it will actually call the macro evaluator. I put all of that logic in this class.